src/features/filter.ts

```diff
@@ -1,12 +1,32 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { PayloadAction, Slice, createSlice } from '@reduxjs/toolkit';
+import { Status } from '../types/Status';
 
 const initialState = {
   query: '',
-  status: 'all',
+  status: 'all' as Status,
 };
 
-export const filterSlice = createSlice({
+type FilterState = {
+  query: string;
+  status: Status;
+};
+
+export const filterSlice: Slice<FilterState> = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setQuery(state, action: PayloadAction<string>) {
+      return {
+        ...state,
+        query: action.payload,
+      };
```

Comment:
By the way,  Redux Toolkit's createSlice uses Immer under the hood, which allows you to mutate the state directly without needing to return a new object.
```suggestion
      state.query = action.payload;
    },
```